File: server/src/models/contact_information.rs
use serde::Serialize;
use sqlx::FromRow;

#[derive(Clone, FromRow, Debug, Serialize)]
pub struct ContactInformationModel {
    pub id: i32,
    pub user_id: i32,
    pub type_field: String,
    pub value: String,
    pub created_at: String,
}

#[derive(Clone, FromRow, Debug, Serialize)]
pub struct PublicContactInformationModel {
    pub type_field: String,
    pub value: String,
}

#[derive(Clone, FromRow, Debug, Serialize)]
pub struct AuthContactInformationModel {
    pub id: i32,
    pub type_field: String,
    pub value: String,
}

File: src/types/contactInformation.ts
export interface ContactInformationModel {
  id: number;
  user_id: number;
  type_field: string;
  value: string;
  created_at: string;
}

export interface PublicContactInformationModel {
  type_field: string;
  value: string;
}

export interface AuthContactInformationModel
  extends PublicContactInformationModel {
  id?: number;
}

export interface AddContactInformationPayload {
  contact_type: string;
  value: string;
}
